Userland/Utilities/tee.cpp

```diff
@@ -26,53 +27,40 @@ static Vector<int> collect_fds(Vector<const char*> paths, bool append, bool* err
     }
 
     Vector<int> fds;
-    for (const char* path : paths) {
-        int fd = open(path, oflag, mode);
-        if (fd < 0) {
-            perror("failed to open file for writing");
-            *err = true;
-        } else {
-            fds.append(fd);
-        }
+    TRY(fds.try_ensure_capacity(paths.size() + 1));
+    for (auto path : paths) {
+        int fd = TRY(Core::System::open(path, oflag, mode));
+        fds.unchecked_append(fd);
     }
-    fds.append(STDOUT_FILENO);
+    fds.unchecked_append(STDOUT_FILENO);
     return fds;
 }
 
-static void copy_stdin(Vector<int>& fds, bool* err)
+static ErrorOr<void> copy_stdin(Vector<int>& fds, bool* err)
 {
     for (;;) {
-        char buf[4096];
-        ssize_t nread = read(STDIN_FILENO, buf, sizeof(buf));
+        Array<u8, 4096> buffer;
+        auto buffer_span = buffer.span();
+        auto nread = TRY(Core::System::read(STDIN_FILENO, buffer_span));
+        buffer_span = buffer_span.trim(nread);
         if (nread == 0)
             break;
-        if (nread < 0) {
-            perror("read() error");
-            *err = true;
-            // a failure to read from stdin should lead to an early exit
-            return;
-        }
 
         Vector<int> broken_fds;
         for (size_t i = 0; i < fds.size(); ++i) {
             auto fd = fds.at(i);
-            int twrite = 0;
-            while (twrite != nread) {
-                ssize_t nwrite = write(fd, buf + twrite, nread - twrite);
-                if (nwrite < 0) {
-                    if (errno == EINTR) {
-                        continue;
-                    } else {
-                        perror("write() failed");
-                        *err = true;
-                        broken_fds.append(fd);
-                        // write failures to a successfully opened fd shall
-                        // prevent further writes, but shall not block writes
-                        // to the other open fds
-                        break;
-                    }
+            auto nwrite_or_error = Core::System::write(fd, buffer_span);
+            if (nwrite_or_error.is_error()) {
+                if (nwrite_or_error.error().code() == EINTR) {
+                    continue;
                 } else {
-                    twrite += nwrite;
+                    perror("write() failed");
```

Comment:
Since you have an `Error` here, let's write it out with `warnln()` instead of using the old-school `perror()`. :)

Reply:
Done! :)